### PR TITLE
Save and Load feature iteration

### PR DESCRIPTION
**Pull request by TeakayAero**
- Added loading from youtube playlist feature
- Added "Also saved for default" feature

### CODE REVIEW

File: src/utils/utils.ts [hunk 1]
```diff
@@ -3,6 +3,9 @@ import * as youtubedl from "youtube-dl";
 
 export const getInfo = Promise.promisify(youtubedl.getInfo);
 
+export const getInfoWithArg = Promise.promisify(
+    (url: string, arg: string[], cb: any) => youtubedl.getInfo(url, arg, cb));
```
Comment (JyaouShingan) [inline src/utils/utils.ts:7]: why callback is `any` not `(Error, Object): void`?

File: src/commands/load.ts [hunk 2]
```diff
@@ -51,4 +54,27 @@ export class LoadCommand extends BaseCommand {
...
     }
+
+    private async loadYoutubeList(
+        message: Message, 
+        guild: GuildManager, 
+        youtubeListUrl: string,
+        collection?: string,
+    ): Promise<void> {
+        const result = await getInfoWithArg(youtubeListUrl, ['--flat-playlist']);
+        message.reply("Start loading from youtube playlist, please be patient...");
```
Comment (JyaouShingan) [inline src/commands/load.ts]: Put this after the array check

File: src/commands/load.ts [hunk 3]
```diff
@@ -51,4 +54,27 @@ export class LoadCommand extends BaseCommand {
...
+        if (Array.isArray(result)) {
+            for (const song of result) {
+                this.logger.info(`Now loading song: ${song.url}`);
+                if (!song.url) continue;
+                const info = await getInfo(song.url);
+                await guild.datasource.saveToPlaylist(
+                    BilibiliSong.withInfo(info, message.author), message.author, collection
+                );
```
Comment (JyaouShingan) [inline src/commands/load.ts:74]: was parallel execution problematic?

**Review by JyaouShingan — CHANGES_REQUESTED** [review]

File: src/utils/utils.ts (hunk 1 above)
Comment (TeakayAero) [reply src/utils/utils.ts:7]: After promisify, the cb field should never be called in real application

File: src/commands/load.ts (hunk 2 above)
Comment (TeakayAero) [reply src/commands/load.ts]: You smart

File: src/commands/load.ts (hunk 3 above)
Comment (TeakayAero) [reply src/commands/load.ts:74]: Want to keep the order of original playlist

Comment (JyaouShingan) [reply src/commands/load.ts:74]: The database does not order it anyway, cuz we don't save timestamp yet

**Review by JyaouShingan — APPROVED** [review]

**Pull request merged**